gammapy/image/maps.py

```diff
@@ -387,6 +387,49 @@ def cutout(self, position, size):
         skymap = SkyMap(data=cutout.data, wcs=cutout.wcs, unit=self.unit)
         return skymap
 
+    def pad(self, factor, mode, **kwargs):
+        """
+        Pad image to the nearest larger shape, that is divisible by the given factor in both axis.
+
+        Calls `numpy.pad`, passing ``mode`` and ``kwargs`` to it.
+
+        Parameters
+        ----------
+        factor : int
+            Factor used for output shape computation
+        mode : str
+            Padding mode, passed to `numpy.pad`.
+
+        Returns
+        -------
+        image : `~gammapy.image.SkyMap`
+            Padded image
+
+        Examples
+        --------
+
+        >>> from gammapy.image import SkyMap
+        >>> image = SkyMap.empty(nxpix=10, nypix=13)
+        >>> print(image.data.shape)
+        (13, 10)
+        >>> image2 = image.pad(factor=4, mode='reflect')
+        >>> image2.data.shape
+        (16, 12)
+        """
+        pad_width = factor - (np.array(self.data.shape) % factor)
+        pad_width = [(0, pad_width[0]), (0, pad_width[1])]
```

Comment:
> But I can make a separate issue for that as a proposed improvement.

That would be my preference.  The plan is that @OlgaVorokh finishes her open PRs, we rename `SkyMap` to `SkyImage`, and then we do the Gammapy 0.5 release in the next days.

Please make separate issues or pull requests for extensions.